contrib/CMakeLists.txt

```diff
@@ -67,17 +69,23 @@ add_contrib (brotli-cmake brotli)
 add_contrib (double-conversion-cmake double-conversion)
 add_contrib (croaring-cmake croaring)
 add_contrib (zstd-cmake zstd)
+set_original_launchers_if_needed()
 add_contrib (zlib-ng-cmake zlib-ng)
```

Comment:
Is it necessary for contribs to be added in a specific order or could we rearrange them to have only one section with original launcher?

Reply:
I asked myself the same, but didn't re-arrange them out of fear that something depended on something else. They shouldn't, but who knows. I'll give it a go, though.

Reply:
Well, surprise surprise... with my attempt to re-arrange all the ones that we need to compile with original linkers at the very beginning gave me a linking error in all builds. e.g. https://s3.amazonaws.com/clickhouse-builds/PRs/73647/10ffe578dd3d2cd7159fbf0cdd7614dc861d3390/package_release/build_log.log

```
Dec 23 18:59:58 [13101/13125] Linking CXX executable programs/clickhouse
Dec 23 18:59:58 FAILED: programs/clickhouse 
Dec 23 18:59:58 : && /usr/bin/clang++-19 --target=x86_64-linux-gnu --sysroot=/build/cmake/linux/../../contrib/sysroot/linux-x86_64/x86_64-linux-gnu/libc --gcc-toolchain=/build/cmake/linux/../../contrib/sysroot/linux-x86_64 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -Wno-enum-constexpr-conversion -fsized-deallocation  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=/build=. -ftime-trace -falign-functions=32 -mbranches-within-32B-boundaries -ffp-contract=off  -fdiagnostics-absolute-paths -fstrict-vtable-pointers -Wall -Wextra -Wframe-larger-than=65536 -Weverything -Wpedantic -Wvla-cxx-extension -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -Wno-enum-constexpr-conversion -Wno-unsafe-buffer-usage -Wno-switch-default -O2 -g -DNDEBUG -O3 -g  -flto=thin -fwhole-program-vtables -fno-pie --gcc-toolchain=/build/cmake/linux/../../contrib/sysroot/linux-x86_64 --ld-path=/usr/bin/ld.lld-19 -Wl,--no-export-dynamic -Wl,--gc-sections -Wl,--build-id=sha1 -Wl,--time-trace -no-pie -Wl,-no-pie -Wl,--allow-multiple-definition  -flto=thin -fwhole-program-vtables   -Xlinker --no-undefined @CMakeFiles/clickhouse.rsp -o programs/clickhouse && cd /build/build_docker/programs && /build/utils/check-style/check-large-objects.sh /build/build_docker && cd /build/build_docker/programs && mkdir -p /build/build_docker/programs/stripped/lib/debug/bin && mkdir -p /build/build_docker/programs/stripped/bin && cp clickhouse /build/build_docker/programs/stripped/bin/clickhouse && /usr/bin/llvm-objcopy-19 --only-keep-debug /build/build_docker/programs/stripped/bin/clickhouse /build/build_docker/programs/stripped/lib/debug/bin/clickhouse.debug && chmod 0644 /build/build_docker/programs/stripped/lib/debug/bin/clickhouse.debug && /usr/bin/llvm-strip-19 --strip-debug --remove-section=.comment --remove-section=.note /build/build_docker/programs/stripped/bin/clickhouse && /usr/bin/llvm-objcopy-19 --add-gnu-debuglink /build/build_docker/programs/stripped/lib/debug/bin/clickhouse.debug /build/build_docker/programs/stripped/bin/clickhouse
Dec 23 18:59:58 ld.lld-19: error: undefined symbol: grpc::SslServerCredentials(grpc::SslServerCredentialsOptions const&)
Dec 23 18:59:58 >>> referenced by GRPCServer.cpp:150 (./build_docker/./src/Server/GRPCServer.cpp:150)
Dec 23 18:59:58 >>>               src/clickhouse.lto.libdbms.a(GRPCServer.cpp.o at 2341790388).o:(DB::GRPCServer::start())
Dec 23 18:59:58 clang++-19: error: linker command failed with exit code 1 (use -v to see invocation)
Dec 23 18:59:58 ninja: build stopped: subcommand failed.
```

Of course, it happened because I added these ones at the very top. I need to add OpenSSL first, because otherwise it'll use the one from the system :upside_down_face: